artifactory/utils/container/buildinfo.go

```diff
@@ -283,145 +127,35 @@ func writeLayersToFile(layers []utils.ResultItem) (filePath string, err error) {
-		log.Debug(`failed to unmarshal fat-manifest`)
-		return "", err
-	}
-	imageOs, imageArch, err := builder.containerManager.OsCompatibility(builder.image)
-	if err != nil {
-		return "", err
-	}
-	return searchManifestDigest(imageOs, imageArch, fatManifestContent.Manifests), nil
+func getManifestArtifact(manifest *utils.ResultItem) (artifact buildinfo.Artifact) {
+	return buildinfo.Artifact{Name: "manifest.json", Type: "json", Checksum: &buildinfo.Checksum{Sha1: manifest.Actual_Sha1, Md5: manifest.Actual_Md5}, Path: path.Join(manifest.Path, manifest.Name)}
 }
-
-// Verify manifest contains the builder image digest. If there is no match, return nil.
-func verifyManifestBySha256(manifestSearchResult utils.ResultItem, builder *buildInfoBuilder) (imageManifest *manifest, err error) {
-	if manifestSearchResult.GetProperty("docker.manifest.digest") != builder.manifestSha256 {
-		log.Debug(`Found incorrect manifest.json file. Expects sha256 "` + builder.manifestSha256 + `" found "` + manifestSearchResult.GetProperty("sha256"))
-		return
-	}
-	log.Debug(`Found manifest.json with expected sha256: "` + builder.manifestSha256)
-	if err = builder.downloadLayer(manifestSearchResult, &imageManifest); err != nil || imageManifest == nil {
-		return
-	}
-	builder.imageId = imageManifest.Config.Digest
-	return
+// Return - fat manifest artifacts as buildinfo.Artifact struct.
+func getFatManifestArtifact(fatManifest *utils.ResultItem) (artifact buildinfo.Artifact) {
+	return buildinfo.Artifact{Name: "list.manifest.json", Type: "json", Checksum: &buildinfo.Checksum{Sha1: fatManifest.Actual_Sha1, Md5: fatManifest.Actual_Md5}, Path: path.Join(fatManifest.Path, fatManifest.Name)}
 }
 
-// Verify manifest by comparing config digest, which references to the image digest. If there is no match, return nil.
-func verifyManifestByDigest(manifestSearchResult utils.ResultItem, builder *buildInfoBuilder) (imageManifest *manifest, err error) {
-	if err = builder.downloadLayer(manifestSearchResult, &imageManifest); err != nil {
-		return
-	}
-	if imageManifest.Config.Digest != builder.imageId {
-		log.Debug(`Found incorrect manifest.json file. Expects digest "` + builder.imageId + `" found "` + imageManifest.Config.Digest)
-		imageManifest = nil
-	}
-	return
+// Return - manifest dependency as buildinfo.Dependency struct.
+func getManifestDependency(searchResults *utils.ResultItem) (dependency buildinfo.Dependency) {
+	return buildinfo.Dependency{Id: "manifest.json", Type: "json", Checksum: &buildinfo.Checksum{Sha1: searchResults.Actual_Sha1, Md5: searchResults.Actual_Md5}}
 }
 
 // Read the file which contains the following format: 'IMAGE-TAG-IN-ARTIFACTORY'@sha256'SHA256-OF-THE-IMAGE-MANIFEST'.
 func GetImageTagWithDigest(filePath string) (tag string, sha256 string, err error) {
+	var buildxMetaData buildxMetaData
 	data, err := ioutil.ReadFile(filePath)
 	if err != nil {
 		log.Debug("ioutil.ReadFile failed with '%s'\n", err)
 		err = errorutils.CheckError(err)
```

Comment:
This line can be removed, and instead, we can modify the above if statement from - 
```
if err != nil {
```
to - 
```
if errorutils.CheckError(err) != nil {
```